pysindy/feature_library/base.py

```diff
@@ -499,20 +497,12 @@ def get_feature_names(self, input_features=None):
                     _unique(self.inputs_per_library[i])
                 ].tolist()
             lib_i_feat_names = lib_i.get_feature_names(input_features_i)
-            for j in range(i + 1, len(self.libraries)):
-                lib_j = self.libraries[j]
-                if input_features is None:
-                    input_features_j = [
-                        "x%d" % k for k in _unique(self.inputs_per_library[j])
-                    ]
-                else:
-                    input_features_j = np.asarray(input_features)[
-                        _unique(self.inputs_per_library[j])
-                    ].tolist()
-                lib_j_feat_names = lib_j.get_feature_names(input_features_j)
-                feature_names += self._name_combinations(
-                    lib_i_feat_names, lib_j_feat_names
-                )
+
+            if len(feature_names) == 0:
+                feature_names = lib_i_feat_names
+            else:
+                feature_names = self._name_combinations(feature_names, lib_i_feat_names)
+
```

Comment:
Similar to above comments:
1. Love how you removed the nested iteration!  Thank you.
2. Set `input features` once above loop instead of checking if it's `None` at each iteration and setting `input_features_i`.
3. Initialize `feature_names` with the result of applying the first library, then iterate through the rest of the libraries, as above.
4. Could you please add static typing to this function signature?
```
    def get_feature_names(self, input_features: list[str] | None=None) -> list[str]:
```